plugins/scanners/fossid/src/main/kotlin/FossIdScanResults.kt

```diff
@@ -326,6 +326,46 @@ private fun getLicenseFindingFromSnippetChoice(
     }
 }
 
+/**
+ * Check all [markedAsIdentifiedFiles] if their snippet choices locations count or non-relevant snippets locations count
+ * matches the ones stored in the [OrtComment]: When not, it means some of this configuration has been removed and the
+ * files should be considered as pending again. Such files are returned.
+ */
+internal fun listUnmatchedSnippetChoices(
+    markedAsIdentifiedFiles: List<MarkedAsIdentifiedFile>,
+    snippetChoices: List<SnippetChoice>
+): List<String> =
+    markedAsIdentifiedFiles.filterNot { markedAsIdentifiedFile ->
+        val markedFileName = markedAsIdentifiedFile.getFileName()
+        val snippetChoicesByName = snippetChoices.filter {
+            it.given.sourceLocation.path == markedFileName
+        }
+        val comment = markedAsIdentifiedFile.comments.values.firstOrNull {
+            it.comment.contains(ORT_NAME)
+        }?.runCatching {
+            jsonMapper.readValue(this.comment, OrtComment::class.java)
+        }?.onFailure {
+            logger.warn {
+                "File $markedFileName is marked as identified but it does have a valid comment ('${it.message}'). " +
+                    "It will be ignored."
+            }
```

Comment:
This seems wrong, `it` is the exception if `readValue` failed, not the `OrtComment`. Or should it say "does **not** have a valid comment"?

Reply:
Yes I missed the _not_. I reworded the log message too.